R/mod_packageDependencies.R

```diff
@@ -0,0 +1,323 @@
+#' @importFrom formattable as.datatable formattable formatter style
+#' @importFrom cli cli_progress_message cli_progress_bar cli_progress_update cli_progress_done
+#' 
+#' @keywords internal
+#' 
+packageDependenciesServer <- function(id, selected_pkg, user, changes, parent) {
+  moduleServer(id, function(input, output, session) {
+    ns <- session$ns
+    
+    metric_wts_df <- dbSelect("SELECT id, name, weight FROM metric")
+    metric_weights <- metric_wts_df$weight
+    names(metric_weights) <- metric_wts_df$name
+    
+    get_versnScore <- function(pkg_name, id) {
+      
+      riskmetric_assess <-
+        riskmetric::pkg_ref(pkg_name,
+                            source = "pkg_cran_remote",
+                            repos = c("https://cran.rstudio.com")) %>%
+        dplyr::as_tibble() %>%
+        riskmetric::pkg_assess()
+      
+      riskmetric_score <-
+        riskmetric_assess %>%
+        riskmetric::pkg_score(weights = metric_weights)
+      
+      cli::cli_progress_update(id = id)
+      
+      return(list(name = riskmetric_assess$package, version = riskmetric_assess$version, score = round(riskmetric_score$pkg_score,2) ))
+    }
+    
+    # used for adding action buttons to data_table
+    shinyInput <- function(FUN, len, id, ...) {
+      inputs <- character(len)
+      for (i in seq_len(len)) {
+        inputs[i] <- as.character(FUN(paste0(id, i), ...))
+      }
+      inputs
+    }
+    
+    depends <- reactive({
+      req(selected_pkg$name() != "-")
+      riskmetric::pkg_ref(selected_pkg$name()) %>% 
+        riskmetric::assess_dependencies() 
+    })
+    
+    revdeps <- reactive({
+      req(selected_pkg$name() != "-")
+      riskmetric::pkg_ref(selected_pkg$name()) %>% 
+        riskmetric::assess_reverse_dependencies() 
+    })
+    
+    dd <- reactive({
+      req(pkg_df())
+      x <- deepdep::deepdep(selected_pkg$name(), depth = 2, dependency_type = c("Depends", "Imports", "LinkingTo"))
+      if (nrow(x) == 0) {
+        x <- data.frame(origin = selected_pkg$name(),
+                        name = pull(pkg_df()[1, 3]), version = NA_character_, type = "Imports",
+                        origin_level = 0L, dest_level = 1L)
+        attributes(x)$class <- c("deepdep", "data.frame")
+        attributes(x)$package_name <- selected_pkg$name()
+      }
+      return(x)
+    })
+    
+    tabready <- reactiveVal()
+    add_pkg <- reactiveVal()
+    
+    observeEvent(parent$input$tabs, {
+      tabready(0L)
+      if(parent$input$tabs == "Package Dependencies") {
+        tabready(1L)
+      }
+      
+    })
+    
+    m_id <- reactiveVal()
+    # start the progress message as soon as possible
+    observeEvent(list(selected_pkg$name(),tabready()), {
+      req(selected_pkg$name() != "-")
+      req(tabready() == 1L)
+      # this is really where the progress bar should start
+      m_id(cli::cli_progress_message("About to Create Data Table...", .auto_close = FALSE))
```

Comment:
This progress message appears and never closes without me having to manually close it.